docs: rework md book (#215)
quay/claircore: docs/reference/versioned_scanner.md

# Versioned Scanner

> A versioned scanner is typically embedded into other scanner types.
> Drives ClairCore's ability to register and understand when updaters have been changed.
> Methods and functions which want to work with a generic scanner type may take a VersionedScanner interface as their arguments.
> 
> Implementors of this interface *must* providea unique name.
> Making changes to an implemented scanner *must* return a new Version.
> Implementors *must* return the kind: "package", "distribution", "repository"

Reviewer: ```suggestion
A versioned scanner is typically embedded into other scanner types.
It drives ClairCore's ability to register and understand when updaters have been changed.
Functions that want to work with a generic scanner type should use a VersionedScanner .

Implementers of this interface *must* provide a unique name.
Making changes to a scanner's implementation *must* return a new Version.
Implementers *must* return the correct kind, one of "package", "distribution", "repository"
```
